### PR TITLE
Hoogle from snapshot

### PR DESCRIPTION
**Pull request by qrilka**
Note: Documentation fixes for https://docs.haskellstack.org/en/stable/ should target the "stable" branch, not master.

Please include the following checklist in your PR:

* [x] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

Tested locally
@snoyberg this also fixes misplaced changelog from #4958

### CODE REVIEW

File: src/Stack/Hoogle.hs [hunk 1]
```diff
@@ -75,49 +80,64 @@ hoogleCmd (args,setup,rebuild,startServer) =
...
+            Just hoogleDep ->
+              case dpLocation hoogleDep of
+                PLImmutable pl -> do
+                  T.pack . packageIdentifierString <$>
+                      restrictMinHoogleVersion muted (packageLocationIdent pl)
+                PLMutable _ -> do
+                  muteableLogInfo muted "Local hoogle package found, installing it"
+                  pure $ T.pack (packageNameString hooglePackageName)
+            Nothing -> do
+              mpir <- getLatestHackageVersion YesRequireHackageIndex hooglePackageName UsePreferredVersions
```
Comment (snoyberg) [inline src/Stack/Hoogle.hs]: What do you think about emitting a warning here to inform that grabbing the latest Hoogle may not work?

File: src/Stack/Hoogle.hs [hunk 2]
```diff
@@ -75,49 +80,64 @@ hoogleCmd (args,setup,rebuild,startServer) =
...
+                { boptsCLITargets =  [hoogleTarget]
                 }
         runRIO config $ withEnvConfig NeedTargets boptsCLI f
+    restrictMinHoogleVersion
+      :: HasLogFunc env
+      => Muted -> PackageIdentifier -> RIO env PackageIdentifier
+    restrictMinHoogleVersion muted ident = do
+      if ident < hoogleMinIdent
+      then do
+          muteableLogInfo muted $
```
Comment (snoyberg) [inline src/Stack/Hoogle.hs]: I think a warning may make more sense here too: this is a situation that's likely to result in failure.

**Comment by qrilka** [comment]
I've added warnings and also changed dealing with mutable location of Hoogle

**Comment by qrilka** [comment]
Rebase resolved it, ok to merge?

**Pull request merged**